tools/twix/src/panels/remote.rs

```diff
@@ -0,0 +1,89 @@
+use std::sync::Arc;
+
+use eframe::egui::Widget;
+use gilrs::{Axis, Button, Gamepad, GamepadId, Gilrs};
+use serde_json::{json, Value};
+use types::Step;
+
+use crate::{nao::Nao, panel::Panel};
+
+pub struct RemotePanel {
+    nao: Arc<Nao>,
+    gilrs: Gilrs,
+    active_gamepad: Option<GamepadId>,
+    enabled: bool,
+}
+
+impl Panel for RemotePanel {
+    const NAME: &'static str = "Remote";
+
+    fn new(nao: Arc<Nao>, _value: Option<&Value>) -> Self {
+        let gilrs = Gilrs::new().unwrap();
+        let active_gamepad = None;
+        let enabled = false;
+
+        Self {
+            nao,
+            gilrs,
+            active_gamepad,
+            enabled,
+        }
+    }
+
+    fn save(&self) -> Value {
+        json!({})
+    }
+}
+
+fn get_axis_value(gamepad: Gamepad, axis: Axis) -> Option<f32> {
+    Some(gamepad.axis_data(axis)?.value())
+}
+
+impl RemotePanel {
+    fn update_step(&self, step: Value) {
+        self.nao
+            .update_parameter_value("control.step_planner.injected_step", step);
+    }
+}
+
+impl Widget for &mut RemotePanel {
+    fn ui(self, ui: &mut eframe::egui::Ui) -> eframe::egui::Response {
+        self.gilrs.inc();
+
+        if ui.checkbox(&mut self.enabled, "Enabled (Start)").changed() {
```

Comment:
Maybe that`s just personal, but for readability I'd prefer
```rs
ui.checkbox(&mut self.enabled, "Enabled (Start)");
if self.enabled { ... }
```

Reply:
That's not the same logic though.
The if block is supposed executed when the enabled state changed, not every render cycle when enabled is true.